### PR TITLE
docs(plugins): updates the description for app-reg

### PR DESCRIPTION
**Pull request by raoxiaoyan**
### Testing instructions

Preview link: 

### Checklist 

- [x] Review label added 
- [ ] [Conditional version tags](https://docs.konghq.com/contributing/conditional-rendering/#conditionally-render-content-by-version) added, if applicable.

For example, if this change is for an upcoming 3.6 release, enclose your content in `{% if_version gte:3.6.x %} <content> {% endif_version %}` tags (or `if_plugin_version` tags for plugins). 

Use any of the following keys:
* `gte:<version>` - greater than or equal to a specific version
* `lte:<version>` - less than or equal to a specific version
* `eq:<version>` - exactly equal to a specific version

You can do the same for older versions.

FTI-5789
KM-47

### CODE REVIEW

**Comment by raoxiaoyan** [comment]
Please wait for this PR to be merged. https://github.com/Kong/kong-ee/pull/8556

**Comment by lena-larionova** [comment]
Why is there development on the app registration plugin in 3.7, if we are no longer supporting the Dev Portal?

**Comment by raoxiaoyan** [comment]
> ~Why is there development on the app registration plugin in 3.7, if we are no longer supporting the Dev Portal?~
> 
> Got my answer in Slack. Adding this change is fine, we'll just need to make some adjustments to the plugin doc itself (outside the scope of this PR).

Thanks. This PR will be backported to 3.4.x, 3.5.x, and 3.6.x.

File: app/_hub/kong-inc/application-registration/how-to/_index.md
Comment (nekolab) [inline app/_hub/kong-inc/application-registration/how-to/_index.md:40]: Activating application registration will restrict access to the service using consumer credentials. To regain this functionality, you can enable the `enable_proxy_with_consumer_credential` flag.

File: app/_hub/kong-inc/application-registration/how-to/_index.md
Comment (nekolab) [inline app/_hub/kong-inc/application-registration/how-to/_index.md:85]: When `enable_proxy_with_consumer_credential` is enabled, constructing a request to the service should yield a 200 response, rather than the 401 response when disabled.

File: app/_hub/kong-inc/application-registration/how-to/_index.md
Comment (nekolab) [inline app/_hub/kong-inc/application-registration/how-to/_index.md:91]: The route should still be accessible with the `client_id` of the application.

File: app/_hub/kong-inc/application-registration/how-to/_index.md
Comment (sumimakito) [inline app/_hub/kong-inc/application-registration/how-to/_index.md:42]:
```suggestion
By enabling the `application-registration` plugin on a service, access to the service with consumer credentials will be disallowed by default. To allow this behavior, you can set `enable_proxy_with_consumer_credential` to `true`.

{:.important}
> **Important:** Please always use `enable_proxy_with_consumer_credential` with caution, as this allows accessing the scoped service via consumer credentials other than your application's client ID (API key).
```

File: app/_hub/kong-inc/application-registration/overview/_index.md
Comment (sumimakito) [inline app/_hub/kong-inc/application-registration/overview/_index.md:11]:
```suggestion
Application Registration plugin. 

{:.note}
> **Note**: By enabling the `application-registration` plugin on a service, access to the service with consumer credentials will be disallowed by default. To allow this behavior, please refer to [these instructions](/hub/kong-inc/application-registration/how-to/#allow-accessing-the-service-with-consumer-credentials-optional) for advanced configuration.
```

As this is optional for our users, would it be better for us to leave a link for them here?

_For Docs team reviewers_: Could you help check if the anchor in the link above is correct? Thanks!

File: app/_hub/kong-inc/application-registration/how-to/_index.md
Comment (sumimakito) [inline app/_hub/kong-inc/application-registration/how-to/_index.md:39]:
```suggestion
### Allow accessing the service with consumer credentials (Optional)
```

Let's point out that this is optional for users.

**Comment by sumimakito** [comment]
Hello @lena-larionova - I've left some suggestions in the comments to describe these changes. Would you mind giving this PR another look? Thanks!

File: app/_hub/kong-inc/application-registration/how-to/_index.md
Comment (lena-larionova) [inline app/_hub/kong-inc/application-registration/how-to/_index.md:39]:
```suggestion
### Allow accessing the service with consumer credentials (optional)
```

File: app/_hub/kong-inc/application-registration/how-to/_index.md
Comment (lena-larionova) [inline app/_hub/kong-inc/application-registration/how-to/_index.md:41]:
```suggestion
By enabling the Application Registration plugin on a service, access to the service with consumer credentials is disallowed by default. To allow this behavior, you can set `enable_proxy_with_consumer_credential` to `true`.
```

File: app/_hub/kong-inc/application-registration/how-to/_index.md
Comment (lena-larionova) [inline app/_hub/kong-inc/application-registration/how-to/_index.md:44]:
```suggestion
> **Important:** Always use `enable_proxy_with_consumer_credential` with caution, as this allows accessing the scoped service via consumer credentials other than your application's client ID (API key).
```

File: app/_hub/kong-inc/application-registration/how-to/_index.md
Comment (lena-larionova) [inline app/_hub/kong-inc/application-registration/how-to/_index.md:46]:
```suggestion
```sh
```

File: app/_hub/kong-inc/application-registration/how-to/_index.md
Comment (lena-larionova) [inline app/_hub/kong-inc/application-registration/how-to/_index.md:54]:
```suggestion
Replace `<plugin_id>` with the `id` from the existing `application-registration` instance:
```

File: app/_hub/kong-inc/application-registration/how-to/_index.md
Comment (lena-larionova) [inline app/_hub/kong-inc/application-registration/how-to/_index.md:56]:
```suggestion
```sh
```

File: app/_hub/kong-inc/application-registration/how-to/_index.md
Comment (lena-larionova) [inline app/_hub/kong-inc/application-registration/how-to/_index.md:61]:
```suggestion
Then, enable the Key Auth plugin and a route for the service:
```

File: app/_hub/kong-inc/application-registration/how-to/_index.md
Comment (lena-larionova) [inline app/_hub/kong-inc/application-registration/how-to/_index.md:63]:
```suggestion
```sh
```

File: app/_hub/kong-inc/application-registration/how-to/_index.md
Comment (lena-larionova) [inline app/_hub/kong-inc/application-registration/how-to/_index.md:68]:
```suggestion
```sh
```

File: app/_hub/kong-inc/application-registration/how-to/_index.md
Comment (lena-larionova) [inline app/_hub/kong-inc/application-registration/how-to/_index.md:74]:
```suggestion
Create a consumer and a credential for the consumer. You'll need this to use key authentication:
```

File: app/_hub/kong-inc/application-registration/how-to/_index.md
Comment (lena-larionova) [inline app/_hub/kong-inc/application-registration/how-to/_index.md:76]:
```suggestion
```sh
```

File: app/_hub/kong-inc/application-registration/how-to/_index.md
Comment (lena-larionova) [inline app/_hub/kong-inc/application-registration/how-to/_index.md:81]:
```suggestion
```sh
```

File: app/_hub/kong-inc/application-registration/how-to/_index.md
Comment (lena-larionova) [inline app/_hub/kong-inc/application-registration/how-to/_index.md:88]:
```suggestion
```sh
```

File: app/_hub/kong-inc/application-registration/how-to/_index.md
Comment (lena-larionova) [inline app/_hub/kong-inc/application-registration/how-to/_index.md:86]:
```suggestion
When `enable_proxy_with_consumer_credential` is enabled, constructing a request to the service should yield a 200 response, rather than the 401 response when disabled:
```

File: app/_hub/kong-inc/application-registration/how-to/_index.md
Comment (lena-larionova) [inline app/_hub/kong-inc/application-registration/how-to/_index.md:92]:
```suggestion
The route should still be accessible with the `client_id` of the application:
```

File: app/_hub/kong-inc/application-registration/how-to/_index.md
Comment (lena-larionova) [inline app/_hub/kong-inc/application-registration/how-to/_index.md:94]:
```suggestion
```sh
```

File: app/_hub/kong-inc/application-registration/overview/_index.md
Comment (lena-larionova) [inline app/_hub/kong-inc/application-registration/overview/_index.md:11]:
```suggestion
> **Note**: By enabling the Application Registration plugin on a service, access to the service with consumer credentials is disallowed by default. To allow this behavior, check the [how-to guide for allowing access to the service](/hub/kong-inc/application-registration/how-to/#allow-accessing-the-service-with-consumer-credentials-optional).
```

**Review by lena-larionova — CHANGES_REQUESTED** [review]

**Comment by lena-larionova** [comment]
This LGTM now, but holding off to merge until it gets into a 3.4 patch. Our doc currently lists 3.4 as the last available version for the plugin.

**Comment by lena-larionova** [comment]
Merge after https://github.com/Kong/kong-ee/pull/8746 goes out + next 3.4 patch after that.

**Comment by raoxiaoyan** [comment]
> Merge after [Kong/kong-ee#8746](https://github.com/Kong/kong-ee/pull/8746) gets merged + next 3.4 patch after that.

@lena-larionova  It is ready for merge.

**Comment by lena-larionova** [comment]
Yep, I've marked this PR for release with the 3.4.3.7 patch. I'll merge when it goes out.

**Review by lena-larionova — APPROVED** [review]

File: app/_hub/kong-inc/application-registration/how-to/_index.md
Comment (lena-larionova) [inline app/_hub/kong-inc/application-registration/how-to/_index.md:39]:
```suggestion
{% if_version gte:3.4.x %}
### Allow accessing the service with consumer credentials (optional)
```

File: app/_hub/kong-inc/application-registration/how-to/_index.md
Comment (lena-larionova) [inline app/_hub/kong-inc/application-registration/how-to/_index.md:97]:
```suggestion
{% endif_version %}
```

File: app/_hub/kong-inc/application-registration/overview/_index.md
Comment (lena-larionova) [inline app/_hub/kong-inc/application-registration/overview/_index.md:12]:
```suggestion
{% if_version gte:3.4.x %}

{:.note}
> **Note**: By enabling the Application Registration plugin on a service, access to the service with consumer credentials is disallowed by default. To allow this behavior, check [how-to guide for allowing access to the service](/hub/kong-inc/application-registration/how-to/#allow-accessing-the-service-with-consumer-credentials-optional) for advanced configuration.

{% endif_version %}
```

**Pull request merged**
